executor/prepared.go

```diff
@@ -222,7 +224,7 @@ func (e *PrepareExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	if !isNoResultPlan(p) {
 		e.Fields = colNames2ResultFields(p.Schema(), p.OutputNames(), vars.CurrentDB)
 	}
-	if e.ID == 0 {
+	if e.ID == 0 && !e.IsGeneralStmt {
 		e.ID = vars.GetNextPreparedStmtID()
 	}
 	if e.name != "" {
```

Comment:
Should we check `!e.IsGeneralStmt` here?

Reply:
You are right. It's updated, PTAL.